pages/give-devices.js

```diff
@@ -21,77 +20,76 @@ const COUNTRIES = [
   { value: 'PH', label: 'Philippines' },
 ];
 
+const formDataInitialState = {
+  name: '',
+  email: '',
+  country: COUNTRIES[0].value,
+  streetAddress: '',
+  streetAddress2: '',
+  city: '',
+  state: '',
+  postalCode: '',
+  phone: '',
+  originalPurchasePrice: '',
+  description: '',
+};
+
+const isEmptyValidator = (value) => value.trim().length !== 0;
+
+const validations = {
+  name: {
+    error: 'Please add your name.',
+    validator: isEmptyValidator,
+  },
+  email: {
+    error: 'Please add your email address.',
+    validator: isEmptyValidator,
+  },
+  streetAddress: {
+    error: 'Please add your street address.',
+    validator: isEmptyValidator,
+  },
+  city: {
+    error: 'Please add your city.',
+    validator: isEmptyValidator,
+  },
+  state: {
+    error: 'Please add your state or province.',
+    validator: isEmptyValidator,
+  },
+  postalCode: {
+    error: 'Please add your postal code.',
+    validator: isEmptyValidator,
+  },
+};
+
 export default function GiveDevicesPage() {
   const router = useRouter();
-  const [modalText, setModalText] = React.useState('');
-  const [isModalOpen, setModalOpen] = React.useState(false);
-  const [name, setName] = React.useState('');
-  const [email, setEmail] = React.useState('');
-  const [country, setCountry] = React.useState(COUNTRIES[0].value);
-  const [streetAddress, setStreetAddress] = React.useState('');
-  const [streetAddress2, setStreetAddress2] = React.useState('');
-  const [city, setCity] = React.useState('');
-  const [state, setState] = React.useState('');
-  const [postalCode, setPostalCode] = React.useState('');
-  const [phone, setPhone] = React.useState('');
-  const [originalPurchasePrice, setOriginalPurchasePrice] = React.useState('');
-  const [description, setDescription] = React.useState('');
+  const [formData, setFormData] = React.useState(formDataInitialState);
+  const [errors, setErrors] = React.useState({});
 
-  const hideModal = () => setModalOpen(false);
-
-  const showModal = (message) => {
-    setModalText(message);
-    setModalOpen(true);
-  };
+  const handleChange = (fieldName) => (value) =>
+    setFormData({ ...formData, [fieldName]: value });
```

Comment:
Whoa, currying and the spread operator, fun stuff! Check this out @RedFox0x20 @ozer619 😄